### PR TITLE
Added report counts to frontpage

### LINKED ISSUE #495: Add report counts to the frontpage
This data is present in the APIs but conditional on the frontend on moderator status. The report coutn is null if you're not a moderator so we can safely show this UI.

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Fixes #495 

#### What's this PR do?
Adds the report counts to the frontpage and removes no longer needed calls to fetch the reports.

#### How should this be manually tested?
Verify that the counts still show up and you don't see report counts for channels you're not a moderator of.

### CODE REVIEW

**Comment by aliceriot** [comment]
works as advertised! reading code now

File: static/js/components/CommentTree.js [hunk 1]
```diff
@@ -130,9 +125,9 @@ export default class CommentTree extends React.Component<*, *> {
               >
                 <a href="#">reply</a>
               </div>}
-            {isModerator && report !== undefined
+            {comment.num_reports
```
Comment (aliceriot) [inline static/js/components/CommentTree.js]: we may want to add a quick test just checking that `Reports: #` doesn't show up when `comment.num_reports` is null

File: static/js/components/CompactPostDisplay_test.js [hunk 2]
```diff
@@ -182,11 +181,12 @@ describe("CompactPostDisplay", () => {
   })
 
   it("should render a report count, if passed a report and moderator", () => {
```
Comment (aliceriot) [inline static/js/components/CompactPostDisplay_test.js]: we might want to fix the test name here

File: static/js/containers/PostPage.js [hunk 3]
```diff
@@ -183,10 +179,6 @@ class PostPage extends React.Component<*, void> {
         )
         moderators = response
       }
```
Comment (aliceriot) [inline static/js/containers/PostPage.js]: we're no longer doing anything with `moderators` here, so I think we can just change this to be

```js
if (!moderators) {
  await dispatch(actions.channelModerators.get(channelName))
}
```

Comment (rhysyngsun) [reply static/js/containers/PostPage.js]: Kinda odd that reassigning a `const` worked in the first place 🤔

**Pull request merged**